internal/commands/push/inputs.go

```diff
@@ -47,17 +49,20 @@ func (i *inputs) Resolve(profile *user.Profile, ui terminal.UI) error {
 		searchPath = profile.WorkingDirectory
 	}
 
+	_, err := os.Stat(path.Join(profile.WorkingDirectory, i.LocalPath))
+	if os.IsNotExist(err) {
+		return errProjectInvalid{Path: i.LocalPath, Exists: false}
+	}
+
 	app, err := local.LoadAppConfig(searchPath)
 	if err != nil {
 		return err
 	}
 
-	if i.LocalPath == "" {
-		if app.RootDir == "" {
-			return errProjectNotFound{}
-		}
-		i.LocalPath = app.RootDir
+	if app.RootDir == "" {
+		return errProjectInvalid{Path: i.LocalPath, Exists: true}
 	}
+	i.LocalPath = app.RootDir
```

Comment:
we can still leave this line wrapped in `if i.LocalPath == "" { ... }`, just so we preserve the user's actual input (if provided)